app/modules/ads/core/stable-ads.js

```diff
@@ -68,15 +68,13 @@ class StableAds {
    */
   init(adsContext = {}, queryParams = {}) {
     const reasonConditionMap = {
-      noexternals_querystring: isQueryParamActive(queryParams.noexternals),
-      noads_querystring: isQueryParamActive(queryParams.noads),
-      mobileapp_querystring: isQueryParamActive(queryParams['mobile-app']),
-      noads_pagetype: adsContext.opts.pageType === 'no_ads',
-      load_failed: this.hasLoadFailed,
+      mobileapp_querystring: isQueryParamActive(queryParams['mobile-app']) ? 'mobileapp_querystring' : false,
+      noads_reasons: adsContext.opts.noAdsReasons,
+      load_failed: this.hasLoadFailed ? 'load_failed' : false,
     };
```

Comment:
Is this ok that we have mixed three types here? string, string[] and boolean. For:
```
{
  foo: 'foo',
  bar: [ '1', '2' ],
  x: false
}
```
we will track: `off_foo,off_1,2`, am I right? Is this correct?
Also, I would go for `string || null` types instead of `string || boolean` combination.

Reply:
Yes, you are right - the same way of tracking is currently on production.
About types - yep, probably `string || null` would be better.

Reply:
Yes, you are right - the same way of tracking is currently on production.
About types - yep, probably `string || null` would be better.